app/modules/ads1115.c

```diff
@@ -225,36 +226,39 @@ static int ads1115_lua_reset(lua_State *L) {
     return 0;
 }
 
-// Initializes ADC
-// Lua:     ads1115.setup(ADDRESS, DEVICE_TYPE)
-static int ads1115_lua_setup(lua_State *L) {
+// Register an ADS device
+// Lua:     ads1115.ADS1115(I2C_ID, ADDRESS)
+static int ads1115_lua_register_1115(lua_State *L) {
+    return ads1115_lua_register(L, ADS1115_ADS1115);
+}
+
+static int ads1115_lua_register_1015(lua_State *L) {
+    return ads1115_lua_register(L, ADS1115_ADS1015);
+}
+
+static int ads1115_lua_register(lua_State *L, uint8_t chip_id) {
     if (!lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
         return luaL_error(L, "wrong arg range");
     }
-    uint8_t i2c_addr = luaL_checkinteger(L, 1);
+    uint8_t i2c_id = luaL_checkinteger(L, 1);
+    if (i2c_id != 0) {
+        return luaL_error(L, "Invalid argument: i2c_id");
```

Comment:
It's recommended to use [`luaL_argcheck()`](http://pgl.yoyo.org/luai/i/luaL_argcheck) when checking arguments.
I understand that you follow the pattern in the existing code, but unfortunately that was already inconsistent.